pkg/controller/erspan_test.go

```diff
@@ -0,0 +1,321 @@
+	return "kube_nfp_static"
+}
+
+func erspanpol(name string, namespace string, dest erspanpolicy.ErspanDestType,
+	source erspanpolicy.ErspanSourceType, labels map[string]string) *erspanpolicy.ErspanPolicy {
+	policy := &erspanpolicy.ErspanPolicy{
+		Spec: erspanpolicy.ErspanPolicySpec{
+			Dest:   dest,
+			Source: source,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Status: erspanpolicy.ErspanPolicyStatus{
+			State: erspanpolicy.Ready,
+		},
+	}
+	var podSelector erspanpolicy.PodSelector
+	podSelector.Namespace = namespace
+	podSelector.Labels = labels
+	policy.Spec.Selector = podSelector
+	return policy
+}
+
+func podifinfodata(name string, namespace string, macaddr string,
+	epg string) *podIfpolicy.PodIF {
+	podifinfo := &podIfpolicy.PodIF{
+		Status: podIfpolicy.PodIFStatus{
+			MacAddr: macaddr,
+			EPG:     epg,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	return podifinfo
+}
+
+func buildSpanObjs(name string, dstIP string, flowID int, adminSt string,
+	dir string, macs []string, vpcs []string) bool {
+
+	srcGrp := apicapi.NewSpanVSrcGrp(name)
+	srcGrp.SetAttr("adminSt", adminSt)
+	apicSlice := apicapi.ApicSlice{srcGrp}
+	srcName := name + "_Src"
+	src := apicapi.NewSpanVSrc(srcGrp.GetDn(), srcName)
+	src.SetAttr("dir", dir)
+	srcGrp.AddChild(src)
+	for _, mac := range macs {
+		fvCEpDn := fmt.Sprintf("uni/tn-%s/ap-%s/epg-%s/cep-%s",
+			"consul", "test-ap", "default", mac)
+		srcCEp := apicapi.NewSpanRsSrcToVPort(src.GetDn(), fvCEpDn)
+		src.AddChild(srcCEp)
+	}
+	lbl := apicapi.NewSpanSpanLbl(srcGrp.GetDn(), name)
+	srcGrp.AddChild(lbl)
+
+	destGrp := apicapi.NewSpanVDestGrp(name)
+	destName := name + "_Dest"
+	dest := apicapi.NewSpanVDest(destGrp.GetDn(), destName)
+	destGrp.AddChild(dest)
+	destSummary := apicapi.NewSpanVEpgSummary(dest.GetDn())
+	destSummary.SetAttr("dstIp", dstIP)
+	destSummary.SetAttr("flowId", strconv.Itoa(flowID))
+	dest.AddChild(destSummary)
+	apicSlice = append(apicSlice, destGrp)
+	for _, vpc := range vpcs {
+		accBndlGrp := apicapi.NewInfraAccBndlGrp(vpc)
+		infraRsSpanVSrcGrp := apicapi.NewInfraRsSpanVSrcGrp(vpc, name)
+		accBndlGrp.AddChild(infraRsSpanVSrcGrp)
+		apicSlice = append(apicSlice, infraRsSpanVSrcGrp)
+		infraRsSpanVDstGrp := apicapi.NewInfraRsSpanVDestGrp(vpc, name)
+		accBndlGrp.AddChild(infraRsSpanVDstGrp)
+		apicSlice = append(apicSlice, infraRsSpanVDstGrp)
+	}
+	return false
+
+}
+
+func checkDeleteErspan(t *testing.T, spanTest erspanTest, cont *testAciController) {
+	tu.WaitFor(t, "delete", 500*time.Millisecond,
+		func(last bool) (bool, error) {
+			//span := spanTests[0]
+			key := cont.aciNameForKey("span",
+				spanTest.erspanPol.Namespace+"_"+spanTest.erspanPol.Name)
+			if !tu.WaitEqual(t, last, 0,
+				len(cont.apicConn.GetDesiredState(key)), "delete") {
+				return false, nil
+			}
+			return true, nil
+		})
+}
+
+func addpods(cont *testAciController, incIps bool, ips []string, ipv4 bool) {
```

Comment:
are you not able to refer it from the Network Policy test code, if required move it to common.